dockstore-client/src/main/java/io/dockstore/client/cli/nested/WESLauncher.java

```diff
@@ -0,0 +1,138 @@
+    public ImmutablePair<String, String> executeEntry(String runCommand, File workingDir) throws RuntimeException {
+        runWESCommand(this.originalParameterFile, this.primaryDescriptor, this.zippedEntry);
+
+        //TODO return something better than this? or change the return....
+        return new ImmutablePair<String, String>("", "");
+    }
+
+    /*
+     * File type must match workflow language possible file types
+     * E.g. for CWL workflows the file extension must be cwl, yaml, or yml
+     * Also include json files
+     */
+    protected boolean fileIsCorrectType(File potentialAttachmentFile) {
+        LanguageType potentialAttachmentFileLanguage = abstractEntryClient.checkFileExtension(potentialAttachmentFile.getName()); //file extension could be cwl,wdl or ""
+        if (potentialAttachmentFile.exists() && !potentialAttachmentFile.isDirectory()) {
+            if (potentialAttachmentFileLanguage.equals(this.languageType) || FilenameUtils.getExtension(potentialAttachmentFile.getAbsolutePath()).toLowerCase().equals("json")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    protected void addFilesToWorkflowAttachment(List<File> workflowAttachment, File zippedEntry) {
+        final File tempDir = Files.createTempDir();
+        try {
+            SwaggerUtility.unzipFile(zippedEntry, tempDir);
+        } catch (IOException e) {
+            System.out.println("Could not get files from workflow attachment. Request not sent.");
+            throw new RuntimeException("Unable to get workflow attachment files from zip file", e);
+        }
+
+        // Put file names in workflow attachment list
+        File[] listOfFiles = tempDir.listFiles();
+        for (int i = 0; i < listOfFiles.length; i++) {
+            if (listOfFiles[i].isFile()) {
+                // TODO check file extension (type) only for a local entry; pass all files found in remote entry folder?
+                // There may be confidential or large files that are not needed in a local directory that should
+                // not be sent to a remote endpoint?
+                if (fileIsCorrectType(listOfFiles[i])) {
+                    System.out.println("Adding file " + listOfFiles[i].getName() + " to workflow attachment");
+                    File fileToAdd = new File(tempDir, listOfFiles[i].getName());
+                    workflowAttachment.add(fileToAdd);
+                } else {
+                    System.out.println("File " + listOfFiles[i].getName() + " is not the correct type for the workflow so it will not be "
+                            + "added to the workflow attachment");
+                }
+            } else if (listOfFiles[i].isDirectory()) {
+                System.out.println("Found directory " + listOfFiles[i].getName());
+            }
+        }
+
+        // TODO this doesn't seem to work on my mac
+        // delete the temporary directory when the Java virtual machine exits
```

Comment:
might need to be empty. 
I normally just use https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html#deleteDirectory(java.io.File) (or the version 3 of it actually)

Reply:
Fixed, now using FileUtils.deleteDirectory